spec/requests/sessions_controller_spec.rb

```diff
@@ -55,6 +55,42 @@
       end
     end
 
+    context 'given an INVALID provider' do
+      it 'returns an unauthorized response with JSON' do
+        application = Fabricate(:application)
+        allow(Doorkeeper::Application).to receive(:first).and_return(application)
+
+        user = Fabricate(:user, provider: User.providers[:google_oauth2])
+
+        post '/api/v1/users/sign_in', params: {
+          user: { email: user.email, password: user.password },
+          client_id: 'client_id',
+          client_secret: 'client_secret'
+        }
+
+        expect(response).to have_http_status(:unauthorized)
+        expect(JSON.parse(response.body)['errors'][0]['detail']).to include(I18n.t('devise.failure.invalid', authentication_keys: :email))
```

Comment:
The detail attribute should be a regular text following `JSON:API` spec:
> a human-readable explanation specific to this occurrence of the problem

And the test can be rewritten as:
```suggestion
        expect(JSON.parse(response.body)['errors'][0]['detail']).to eq(I18n.t('devise.failure.invalid', authentication_keys: :email))
```